SU2_CFD/src/driver_adjoint_singlezone.cpp

```diff
@@ -0,0 +1,521 @@
+
+    switch (config->GetKind_Solver()) {
+
+    case DISC_ADJ_EULER: case DISC_ADJ_NAVIER_STOKES: case DISC_ADJ_RANS:
+    case DISC_ADJ_FEM_EULER : case DISC_ADJ_FEM_NS : case DISC_ADJ_FEM_RANS :
+      cout << "log10[U(0)]: "   << log10(solver[FLOW_SOL]->GetRes_RMS(0))
+           << ", log10[U(1)]: " << log10(solver[FLOW_SOL]->GetRes_RMS(1))
+           << ", log10[U(2)]: " << log10(solver[FLOW_SOL]->GetRes_RMS(2)) << "." << endl;
+      cout << "log10[U(3)]: " << log10(solver[FLOW_SOL]->GetRes_RMS(3));
+      if (geometry->GetnDim() == 3) cout << ", log10[U(4)]: " << log10(solver[FLOW_SOL]->GetRes_RMS(4));
+      cout << "." << endl;
+      if ( config->GetKind_Turb_Model() != NONE && !config->GetFrozen_Visc_Disc()) {
+        cout << "log10[Turb(0)]: "   << log10(solver[TURB_SOL]->GetRes_RMS(0));
+        if (solver[TURB_SOL]->GetnVar() > 1) cout << ", log10[Turb(1)]: " << log10(solver[TURB_SOL]->GetRes_RMS(1));
+        cout << "." << endl;
+      }
+      if (config->GetWeakly_Coupled_Heat()){
+        cout << "log10[Heat(0)]: "   << log10(solver[HEAT_SOL]->GetRes_RMS(0)) << "." << endl;
+      }
+      break;
+
+    case DISC_ADJ_FEM:
+
+      if (config->GetGeometricConditions() == LARGE_DEFORMATIONS){
+        cout << "UTOL-A: "   << log10(solver[FEA_SOL]->GetRes_FEM(0))
+             << ", RTOL-A: " << log10(solver[FEA_SOL]->GetRes_FEM(1))
+             << ", ETOL-A: " << log10(solver[FEA_SOL]->GetRes_FEM(2)) << "." << endl;
+      }
+      else{
+        if (geometry->GetnDim() == 2){
+          cout << "log10[RMS Ux]: "   << log10(solver[FEA_SOL]->GetRes_RMS(0))
+               << ", log10[RMS Uy]: " << log10(solver[FEA_SOL]->GetRes_RMS(1)) << "." << endl;
+        }
+        else{
+          cout << "log10[RMS Ux]: "   << log10(solver[FEA_SOL]->GetRes_RMS(0))
+               << ", log10[RMS Uy]: " << log10(solver[FEA_SOL]->GetRes_RMS(1))
+               << ", log10[RMS Uz]: " << log10(solver[FEA_SOL]->GetRes_RMS(2))<< "." << endl;
+        }
+      }
+
+      break;
+
+    case DISC_ADJ_HEAT:
+      cout << "log10[Cons(0)]: "   << log10(solver[HEAT_SOL]->GetRes_RMS(0)) << "." << endl;
+      break;
+
+    }
+
+    cout << "-------------------------------------------------------------------------" << endl << endl;
+  }
+  else if ((rank == MASTER_NODE) && (kind_recording == SecondaryVariables) && (SecondaryVariables != NONE)){
+    cout << endl << "Recording the computational graph with respect to the ";
+    switch (SecondaryVariables){
+      case MESH_COORDS: cout << "mesh coordinates." << endl;    break;
+      default:          cout << "secondary variables." << endl; break;
+     }
+  }
+
+}
+
+void CDiscAdjSinglezoneDriver::MainRecording(){
+
+  /*--- SetRecording stores the computational graph on one iteration of the direct problem. Calling it with NONE
+   *    as argument ensures that all information from a previous recording is removed. ---*/
+
+  SetRecording(NONE);
```

Comment:
Is this required? I ask because it takes time (especially for FEA+Topology) and in the Adj FSI driver I have it commented out and all seems fine.

Reply:
Actually, I have not tested it without, so it might be the case that it is not needed. Let me comment it out and see what it does.

Reply:
I just tested it, flow adjoint tests are all failing 
https://travis-ci.org/su2code/SU2/builds/526434537
so I reverted the changes. I think that @talbring can explain better why that's necessary.

Reply:
Interesting, thanks for trying it. The FSI adjoint seems to work fine without it though, it definitely does not diverge, another AD mystery.